From f77fc35f1929621ffb9527cc29257456cc58f4d8 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 21 Mar 2016 13:09:26 +0800
Subject: [PATCH 0986/5242] MLK-12464-2: ASoC: wm8960: fix clock is not
 correct after suspend/resume

commit  5d34ea66ce6dbb7c57d6051998bb4d7a43e38740 from
https://source.codeaurora.org/external/imx/linux-imx.git

After the suspend/resume, hw_params may be called in bias_level is not
BIAS_ON, then the PLL is not disable/enabled, if the sample rate is
changed, the output clock is not correct.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/wm8960.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/sound/soc/codecs/wm8960.c b/sound/soc/codecs/wm8960.c
index c30f5aa..7b6f709 100644
--- a/sound/soc/codecs/wm8960.c
+++ b/sound/soc/codecs/wm8960.c
@@ -862,8 +862,7 @@ static int wm8960_hw_params(struct snd_pcm_substream *substream,
 
 	wm8960->is_stream_in_use[tx] = true;
 
-	if (snd_soc_component_get_bias_level(component) == SND_SOC_BIAS_ON &&
-	    !wm8960->is_stream_in_use[!tx])
+	if (!wm8960->is_stream_in_use[!tx])
 		return wm8960_configure_clocking(component);
 
 	return 0;
@@ -916,10 +915,6 @@ static int wm8960_set_bias_level_out3(struct snd_soc_component *component,
 				}
 			}
 
-			ret = wm8960_configure_clocking(component);
-			if (ret)
-				return ret;
-
 			/* Set VMID to 2x50k */
 			snd_soc_component_update_bits(component, WM8960_POWER1, 0x180, 0x80);
 			break;
-- 
1.7.9.5

